### DISCUSSION TITLE
Google Summer of Code (GSOC) 2021

### DISCUSSION THREAD

**Discussion by ltalirz** [aiidateam/aiida-core#4702] [Q&A]
After a successful GSOC 2020 under the [NumFocus](https://numfocus.org/) umbrella, the AiiDA team plans to participate in [GSOC 2021](https://summerofcode.withgoogle.com/) as well.

Last year's GSOC student @unkcpz [set the bar pretty high](https://summerofcode.withgoogle.com/archive/2020/projects/5283081100984320/) (and will join [THEOS](http://theossrv1.epfl.ch/) later this year), but don't be shy - there is [lots more to do](https://github.com/aiidateam/aiida-core/wiki/GSoC-2021-Projects)!

This thread is for any discussions leading up to GSOC 2021. If you're interested in working on aiida-core as a student, use this thread to say hi and ask any questions you may have.
Have a look at our [project ideas](https://github.com/aiidateam/aiida-core/wiki/GSoC-2021-Projects) for GSOC 2021, and see the [NumFOCUS guide for GSOC students](https://github.com/numfocus/gsoc/blob/master/CONTRIBUTING-students.md).

Just as a reminder, here is an excerpt from the [GSOC 2021 timeline](https://summerofcode.withgoogle.com/how-it-works/#timeline):

* Organizations Announced: March 9, 2021  
  Interested students can now begin discussing project ideas with accepted mentor organizations.  
  **UPDATE: Great news - NumFOCUS was selected as a GSOC 2021 organization**

* Student Application Period March 29, 2021 - April 13, 2021  
  Students can register and submit their applications to mentor organizations. 

* Student Projects Announced May 17, 2021
  Accepted students are paired with a mentor and start planning their projects and milestones.  

The GSOC 2021 mentors 
Leopold @ltalirz  and Chris @chrisjsewell

**Comment by prinsusinghal** [comment]


**Reply by ltalirz** [reply]
Hey Prateek! 

Happy to see your enthusiasm :-) I certainly don't want to stop you from getting your hands dirty with one of the GSOC projects, but I should mention that GSOC 2021 will [officially start May 17th](https://summerofcode.withgoogle.com/how-it-works/) after student selections are announced.
Until then, time available for mentoring will be quite limited, i.e. you would be a bit on your own.

As mentioned on the [GSOC 2021 ideas page](https://github.com/aiidateam/aiida-core/wiki/GSoC-2021-Projects), a good way to introduce yourself to the team is to take one of the "good first issues" and create a pull request to fix it (or screen the issue tracker for any other issues that you think are up your alley).
This will help you understand the AiiDA codebase as well as help us get a sense of your coding skills - and it is something you can start right away.

Cheers,
Leo

**Reply by prinsusinghal** [reply]
Sure. Thanks @ltalirz

**Comment by kenkirito** [comment]
hi i am interested in this project and I really like to work with you all. how can I get started with my first issue. Thank you for your guidence

**Reply by chrisjsewell** [reply]
Hey @kenkirito!
Yes thanks it looks like you have already identified an issue to try tackling (#4679) so I would just go ahead and give it a go 😄 
There is some guidance for creating PRs at https://github.com/aiidateam/aiida-core/wiki/Making-a-pull-request, and also some guidance on how we review them: https://github.com/aiidateam/aiida-core/wiki/Best-practises-for-code-review. But feel free to ask any further questions here

**Comment by himanshu007-creator** [comment]
Hi @ltalirz sir, I would like to work on the project <strong>Project 4 - Make the AiiDA REST API extensible through plugins</strong>.
Please guide me , how should I make sure i join you this summer? 😀

**Reply by ltalirz** [reply]
Hey @himanshu007-creator! 
As mentioned above, a good way to get started is to take on one of the "good first issues", which will help you get familiar with the AiiDA codebase and help us get a sense of your coding skills.

Student applications start March 29th and successful applications will be announced May 17th.

**Comment by nikJ13** [comment]
Hi @ltalirz ! So, I just wanted to know whether it is necessary to have prior knowledge of material sciences in order to contribute. If not, will there be a necessity to learn material sciences as we progress, working on the project? Or is it alright to just have expertise in the required technical skills?

**Reply by ltalirz** [reply]
Hi @nikJ13 - as mentioned in the [ideas page](https://github.com/aiidateam/aiida-core/wiki/GSoC-2021-Projects), a background in materials science is not needed for the projects, and the focus will be on writing code. Feel free to apply! 

That being said, we do consider a background in materials science a plus - also because it increases the likelihood of AiiDA becoming part of the student's toolbox beyond the the completion of the project.

**Comment by z3y50n** [comment]
Hello @ltalirz I am Dimitris, currently studying Electrical & Computer Engineering in Thessaloniki, Greece and find this project quite interesting. I just wanted to introduce myself before I start working on some first issues in order to get accustomed with the code.

**Comment by lkbhitesh07** [comment]
Hello @chrisjsewell, @ltalirz and everyone, I'm just here to say hello to everyone also I have done reading all the guide you have provided and it's very helpful. I will be getting started with a good first issue also I'm new to all this and I will be seeking your help.

**Reply by chrisjsewell** [reply]
Thanks @lkbhitesh07, looking forward to it 😄

**Comment by spiderxm** [comment]
Hello everyone, I'm Mrigank Anand, I'm looking to contribute.

**Comment by Abiolade1** [comment]
Hi
I am a master's student. I am interested in learning here.

**Comment by pradipchm** [comment]
Hi, 
I am a Ph.D. student at UNT Texas working on computational material science. My group does work on Aiida plugin development. I am quite new to Aiida and last few months, I am trying to learn from tutorials. I am very much interested in GSoC 2021 work for Aiida. I don't know what kind of proposal I need to submit. Could you please guide me regarding the proposal for the application?

**Reply by chrisjsewell** [reply]
Hey @pradipchm, I believe the general guidelines for the proposal are at https://google.github.io/gsocguides/student/writing-a-proposal and also a more specific one for NumFocus here https://github.com/numfocus/gsoc/blob/master/CONTRIBUTING-students.md, but let us know if you have any more specific questions

**Comment by marwanalam2k** [comment]
> After a successful GSOC 2020 under the [NumFocus](https://numfocus.org/) umbrella, the AiiDA team plans to participate in [GSOC 2021](https://summerofcode.withgoogle.com/) as well.
> 
> Last year's GSOC student @unkcpz [set the bar pretty high](https://summerofcode.withgoogle.com/archive/2020/projects/5283081100984320/) (and will join [THEOS](http://theossrv1.epfl.ch/) later this year), but don't be shy - there is [lots more to do](https://github.com/aiidateam/aiida-core/wiki/GSoC-2021-Projects)!
> 
> This thread is for any discussions leading up to GSOC 2021. If you're interested in working on aiida-core as a student, use this thread to say hi and ask any questions you may have.
> 
> Just as a reminder, here is an excerpt from the [GSOC 2021 timeline](https://summerofcode.withgoogle.com/how-it-works/#timeline):
> 
> * Organizations Announced: March 9, 2021
>   Interested students can now begin discussing project ideas with accepted mentor organizations.
>   **UPDATE: Great news - NumFOCUS was selected as a GSOC 2021 organization**
> * Student Application Period March 29, 2021 - April 13, 2021
>   Students can register and submit their applications to mentor organizations.
> * Student Projects Announced May 17, 2021
>   Accepted students are paired with a mentor and start planning their projects and milestones.
> 
> See also the [NumFOCUS guide for GSOC students](https://github.com/numfocus/gsoc/blob/master/CONTRIBUTING-students.md).
> 
> The GSOC 2021 mentors
> Leopold @ltalirz and Chris @chrisjsewell

Hi, I am Marwan Alam. I am interested in joining the Google Summer of Code 2021.
I have done research in computational physics at master level.
I am willing to know about the application procedure.
Thank you.

**Comment by sanjeevthakur123** [comment]
Hello Sir, I am interested in doing this project, but can you give more insight into the project?

**Comment by mohsinm-dev** [comment]
Hi Everyone! My name is mohsin mahmood. Is a Computer science student comfortable with python C C++ kindly tell me from where should I start and kindly tell me am I too late? Any help would be appreciable. I am interested in this project kindly guide me

**Comment by Romakanta** [comment]
Hi there!
I am a Ph.D. student in Computational Physics. I am really interested in this program.

**Comment by gautamsharma480** [comment]
Dear Sir, 

I am a final year Ph.D student at IISER Pune, India. I am working in the field of applications of Density functional theory. I am very much interested in this program. Please include me.

**Reply by gautamsharma480** [reply]
Hi, 

I am interested to develop a fully automated python code based on AiiDA to compute certain physical quantities that are of central importance for the people working in computational material science, particularly, Thermoelectrics. Later on, I want to predict these physical quantities for some unknown materials using Machine Learning.  There a high chance that this project will lead to scientific publication.
Will GSoC (NumFocus) sponsor me for this and provide me a mentor and platform for this project?

**Reply by ltalirz** [reply]
Hey @gautamsharma480 , thanks for asking. Google summer of code is about coding and moving the open source project (here: aiida-core) forward, not about your personal research - for our projects ideas, see the ideas page.
That being said, if you can think of a project that accomplishes both (e.g. by adding a missing feature in AiiDA) that's great - let us know about your project idea in some detail and we'll happily consider it.

As mentioned before, a good way to introduce yourself is to open a PR to fix an existing AiiDA issue, fixing a small bug or making a small improvement.

**Comment by DirectriX01** [comment]
Hello everyone! 
I am Abhinav, and I am looking forward to join you all this summer

**Comment by ltalirz** [comment]
Hello everyone, thanks for checking in!
If you haven't done so already, please have a look at our [project ideas](https://github.com/aiidateam/aiida-core/discussions/4702#discussioncomment-568294) for GSOC 2021.
As mentioned there, a great way to introduce yourself is to take on one of the "good first issues" (or really any issue that interests you and you think could tackle) and make a pull request for it.
Tag @chrisjsewell or me in the PRs and we'll see that it moves forward (sorry if we haven't been too responsive until now; we've had a lot on our plates). We may then contact you personally to get to know you and your background.

Please note that we take our mentoring role seriously and will therefore not be taking more than one GSOC student.
From our student (by the way, all genders equally welcome!) we expect both enthusiasm for one of the projects and a demonstration of your ability. 
Feel free to reach out if you have any further questions.

Cheers,
Leo

**Comment by Seoinage** [comment]
Hello my name is Luis Ponce from Mexico and Im currenlty working with DFT, vasp and wien2k codes to studing electronic and thermoelectric properties I also have knolwdge of python

**Comment by ltalirz** [comment]
Hey guys, this is just a reminder that the student application deadline is April 14th (**edit** or 13th, depending on your timezone, see [timeline](https://summerofcode.withgoogle.com/how-it-works/#timeline)).
We've already reached out to the first student candidates - if you'd like to be considered as well, please consider introducing yourself with a small code contribution to aiida-core as discussed above and think about which [project](https://github.com/aiidateam/aiida-core/wiki/GSoC-2021-Projects) you would be interested and fit to tackle

Cheers,
Leo

**Comment by Joshua-Chris-L** [comment]
Hi i am physics student with physics background both at bachelors and master level. I work with python and very much interested in both computational physics and coding. I will like to be part of this project both to learn and contribute my own quota. 

I will be grateful to be part of this work.

Cheers,
Joshua

**Comment by Compphysguy** [comment]
Good Evening, 
I am Raj Kumar Paudel permanently from Nepal currently doing PhD in Taiwan. I worked in computational physics on the development of density functional theory using mixed basis sets(2D along x and y axis plane wave and bspline along z axis). during learning and training from last 3 years i got somewhat interested in coding. Basically I code quite familiar in fortran and occasionally python, matlab and c. All courses are self taught. I want to join GSOC 2021 to do some work on project. I am looking forward to join this school. Thank you!

**Comment by jurca97edi** [comment]
Hi everyone,
My name is Zsurka Eduárd, I am second year Master Student at ELTE in Budapest. I am studying mainly condensed matter physics and I'm doing my thesis on Cooper pair splitting in graphene junctions. I would like to help developing AIIDA, more precisely the project regarding implementing support for containerized simulation codes seems quite interesting to me. I have plenty of experience with python, since almost all the programming we do at Uni is in python. I am also working part-time at Ericsson Hungary, and I have some experience with docker too. Unfortunately I got word of GSoC quite late and know I am a bit close to the deadline. If there are still places left, I will try to contribute in order to be eligible. 

Best wishes,
Zsurka Eduárd

**Reply by ltalirz** [reply]
Hey @jurca97edi - time is a bit tight but you are welcome to submit a proposal on this project idea.
I can't promise that we'll have time to review it before the deadline, but I'll try to give a look.

Cheers, Leo

**Comment by Seoinage** [comment]
Hello, Luis Ponce from Mexico, and I'm currently working with DFT, vasp,
and wien2k codes to study electronic and thermoelectric properties. I also
have knowledge of python I like to work in the Project 3 - Built-in support
for containerized simulation codes (docker, shifter, singularity, ...)

**Comment by ltalirz** [comment]
Hey guys,

thanks to all for saying hi and in particular to those of you who tried tackling one of the "good first issues" and perhaps even submitted a GSOC proposal! We will reply to each proposal submission individually in due course.
The GSOC submission deadline on April 13th has passed - for those of you who did not submit a proposal but would still like to contribute to the AiiDA ecosystem: code contributions are always welcome, independently of GSOC!
If you have and idea for a new feature that you'd like to work on, or some bug you're trying to fix, always feel free to reach out via the [AiiDA issue tracker](https://github.com/aiidateam/aiida-core/issues) and we'll put you in contact with the right people.

This concludes this discussion thread - I'll leave it open for a couple more days and then lock it.

Have a great weekend!
Leo
